### PR TITLE
Add workflow for running tests on PR

### PR DESCRIPTION
**Pull request by allenporter**
Add a github action for running tests on new PRs and commits.

### CODE REVIEW

**Comment by allenporter** [comment]
Marking as a draft while working getting it to work.

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install pytest
+        pip install -r requirements_dev.txt
+        pip install -r requirements_test.txt
+    - name: Test with pytest
+      run: |
+        pytest tests/
```
Comment (iMicknl) [inline .github/workflows/pytest.yml:29]:
```
  /opt/hostedtoolcache/Python/3.9.10/x64/lib/python3.9/site-packages/_pytest/config/__init__.py:1233: PytestConfigWarning: Unknown config option: asyncio_mode
```

could this be related to your previous PR? Should we remove that mode?

Comment (allenporter) [reply .github/workflows/pytest.yml:29]: Ah it looks like this option is only available for pytest-asyncio>=0.17 https://github.com/pytest-dev/pytest-asyncio#modes -- but maybe its not installed. Let me try adding as an explicit dependency.

Comment (allenporter) [reply .github/workflows/pytest.yml:29]: I went ahead and removed this. The original motivation was for client library tests, before adding the home assistant test harness, so you're right we should just remove this mode. Can add back later if needed.

**Pull request merged**